test/tests/meshmodifiers/ic_for_newlyactivated_nodes/tests

```diff
@@ -0,0 +1,174 @@
+      type = 'Exodiff'
+      input = 'external_moving_boundary_npr_two_variables_only_setIC_for_UO_other_follow_variable.i'
+      exodiff = 'external_moving_boundary_npr_two_variables_only_setIC_for_UO_other_follow_variable_out.e external_moving_boundary_npr_two_variables_only_setIC_for_UO_other_follow_variable_out.e-s002 external_moving_boundary_npr_two_variables_only_setIC_for_UO_other_follow_variable_out.e-s003 external_moving_boundary_npr_two_variables_only_setIC_for_UO_other_follow_variable_out.e-s004'
+      detail = 'The system shall allow different initial condition strategies for each variable on newly activated nodes. In this case, variable `diff` uses nodal patch recovery, while variable `diff2` is set to 10.0.'
+    []
+
+    [IC_for_newlyactivated_nodes_two_variables_same_IC]
+      type = 'Exodiff'
+      input = 'external_moving_boundary_npr_two_variables_npr_for_both.i'
+      exodiff = 'external_moving_boundary_npr_two_variables_npr_for_both_out.e external_moving_boundary_npr_two_variables_npr_for_both_out.e-s002 external_moving_boundary_npr_two_variables_npr_for_both_out.e-s003 external_moving_boundary_npr_two_variables_npr_for_both_out.e-s004'
+      detail = 'The system shall allow a shared initial condition to be applied to multiple variables using a single IC strategy when both are listed in `nodal_patch_recovery_uo`.'
+    []
+
+    [IC_for_newlyactivated_nodes_two_variables_normal_setting]
+      type = 'Exodiff'
+      input = 'external_moving_boundary_npr_two_variables_npr_normal_set.i'
+      exodiff = 'external_moving_boundary_npr_two_variables_npr_normal_set_out.e external_moving_boundary_npr_two_variables_npr_normal_set_out.e-s002 external_moving_boundary_npr_two_variables_npr_normal_set_out.e-s003 external_moving_boundary_npr_two_variables_npr_normal_set_out.e-s004'
+      detail = 'The system shall accept consistent vector sizes for `ic_strategy`, `ic_variables`, and `nodal_patch_recovery_uo` when applying initial conditions.'
+    []
+
+    [npr_element_not_specified_by_user]
+      type = 'Exodiff'
+      input = 'external_moving_boundary_npr_elem_not_provide.i'
+      exodiff = 'external_moving_boundary_npr_elem_not_provide_out.e external_moving_boundary_npr_elem_not_provide_out.e-s002'
+      detail = 'The system shall support nodal patch recovery using all elements in restricted blocks when `use_specific_elements = false`.'
+    []
+
+    [IC_default]
+      type = 'Exodiff'
+      input = 'external_moving_boundary_no_unsolved_blocks_all_IC_default.i'
+      exodiff = 'external_moving_boundary_no_unsolved_blocks_all_IC_default_out.e external_moving_boundary_no_unsolved_blocks_all_IC_default_out.e-s002 external_moving_boundary_no_unsolved_blocks_all_IC_default_out.e-s003 external_moving_boundary_no_unsolved_blocks_all_IC_default_out.e-s004'
+      detail = 'The system shall run with `ic_strategy = IC_DEFAULT` without requiring `nodal_patch_recovery_uo` or `unsolved_blocks`.'
```

Comment:
please dont use parameter names or values. only english in SQA doco, no moose-talk